src/features/movies/moviesSaga.js

```diff
@@ -0,0 +1,36 @@
+import axios from "axios";
+import { takeLatest, put } from "redux-saga/effects";
+import { fetchError, fetchMoviesSuccess, fetchMovieSuccess } from "./moviesSlice";
+import { apiUrl, apiKey } from "../../apiData";
+
+function* fetchMoviesHandler() {
+    try {
+        const data = yield axios.get(
+            `${apiUrl}movie/popular?api_key=${apiKey}&language=en-US&page=1`
+        );
+        const movies = yield data.results;
+        yield put(
+            fetchMoviesSuccess({
+                movies,
+            })
+        );
+    } catch (error) {
+        yield put(fetchError());
+    }
+}
+
+function* fetchMovieHandler({ payload: id }) {
+    try {
+        const movie = yield axios.get(
+            `${apiUrl}movie/${id}?api_key=${apiKey}&language=en-US`
+        );
+        yield put(fetchMovieSuccess({ payload: movie }));
+    } catch (error) {
+        yield put(fetchError());
+    }
+}
+
+export function* watchFetchPopularMovies() {
```

Comment:
there is no import fetchMovies & and fetchMovie from slice, 
Exported fn should have same name as in rootSaga - moviesSaga()

Reply:
this is only example implementation I added comments in root